application_sdk/application/__init__.py

```diff
@@ -192,13 +234,46 @@ async def setup_server(
                     triggers=[event_trigger],
                 )
 
-        # register the workflow on the application server
-        # the workflow is by default triggered by an HTTP POST request to the /start endpoint
+        # Register the main workflow (HTTP POST /start endpoint)
         self.server.register_workflow(
             workflow_class=workflow_class,
             triggers=[HttpWorkflowTrigger()],
         )
 
+    async def _create_mcp_asgi_app(self):
+        """Create MCP server and return ASGI app with proper lifespan."""
```

Comment:
docs: we follow google style guide for docstrings ([refer](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings))